File: cpp/1935.maximum-number-of-words-you-can-type.cpp
/*
 * @lc app=leetcode id=1935 lang=cpp
 *
 * [1935] Maximum Number of Words You Can Type
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    int canBeTypedWords(string text, string brokenLetters)
    {
        string brokenalpha = string(26, 0);
        for (auto a : brokenLetters)
        {
            brokenalpha[a - 'a'] = 1;
        }
        int error_count = 0;
        int res = 0;
        for (auto a : text)
        {
            if (isblank(a))
            {
                if (error_count == 0)
                {
                    ++res;
                }
                error_count = 0;
            }
            else if (error_count > 0)
            {
                continue;
            }
            else if (brokenalpha[a - 'a'])
            {
                ++error_count;
            }
        }
        return res + (error_count == 1 ? 0 : 1);
    }
};
// @lc code=end

File: js/1935.maximum-number-of-words-you-can-type.js
/*
 * @lc app=leetcode id=1935 lang=javascript
 *
 * [1935] Maximum Number of Words You Can Type
 */

// @lc code=start
/**
 * @param {string} text
 * @param {string} brokenLetters
 * @return {number}
 */
var canBeTypedWords = function (text, brokenLetters) {
    let brokenalpha = new Array(26)
    for (let a of brokenLetters) {
        brokenalpha[a.charCodeAt() - 'a'.charCodeAt()] = 1
    }
    let res = 0;
    let error = 0;
    for (let a of text) {
        if (a === ' ') {
            if (error === 0) {
                ++res;
            }
            error = 0;
        } else if (error > 0) {
            continue;
        } else if (brokenalpha[a.charCodeAt() - 'a'.charCodeAt()]) {
            ++error;
        }
    }
    return res + (error === 0 ? 1 : 0);
};
// @lc code=end